src/main/java/http_server/Session.java

```diff
@@ -0,0 +1,50 @@
+package http_server;
+
+public class Session implements Runnable {
+    private final SocketWrapper client;
+    private Router router;
+    private Boolean isRunning;
+
+    public Session(SocketWrapper client, Router router) {
+        this.client = client;
+        this.router = router;
+        this.isRunning = true;
+    }
+
+    public void run() {
+        while (!Thread.interrupted() && isRunning) {
+            if (!client.ready()) {
+                try {
+                    Thread.sleep(1000);
+                    continue;
+                } catch (InterruptedException e) {
+                    close();
+                    return;
+                }
+            }
+
+            String input = client.readData();
+
+            RequestParser parser = new RequestParser(input);
+            Response responder = new Response(parser, router);
+            String response = responder.generateResponse();
```

Comment:
Super small but, you could just throw out this `responder` intermediary entirely:

```
String response = new Response(parser, router).generateResponse()
```